Dynatrace/dynatrace-operator
# Improves makefile and makefile structure

# Description

Improves the Makefile by implementing [best practices](https://docs.cloudposse.com/reference/best-practices/make-best-practices/#use-standard-target-names-in-root-makefile).
Makes the main Makefile more readable and refactors complicated targets into their own files and folders.
This was done to improve readability, usability and maintainability of the Makefile and build system.

## How can this be tested?
Checkout the branch an run your favourite make commands.
Important ones such as `make bundle` and `make manifests` should work as before.
If there was recently work done on any target, it should still run as intended.

## Checklist
~~- [ ] Unit tests have been updated/added~~
Unit tests are not relevant for this PR as it only effects the build system
